src/core/email-client.js

```diff
@@ -53,101 +71,250 @@ function getFreshImap (cb) {
 }
 
 class EmailClient {
-    constructor () {
+    //constructor takes the full name of user as String
+    constructor (name, eventBus) {
         var self = this;
-
-        //  Get Imap obj and check emails
-        getFreshImap(function (err, imap) {
+        self._name = name; 
+        self._email = "";
+        self._imap = Q.defer();
+        self._transporter = Q.defer();
+        self._status = Q.defer();
+        self._SendCheck = Q.defer();
+        self._changeCheck = Q.defer();
+        self._eventBus = eventBus;
+       
+        getFreshImap(function (err, imap, email, smtpConfig) {
             if (err) {
                 console.log(err);
             } else {
-                self.imap = imap;
-                self._checkEmails();
+                self._imap.resolve(imap);
+                self._email = email;
+                self._transporter.resolve(NodeMailer.createTransport(smtpConfig));
+                
             }
         });
     }
 
-    _checkEmails () {
+    _loadInfo(){
+        //  Get Imap obj and check emails
+        var self = this;
+        return self._imap.promise;
+    }
+
+    _loadTransporter(){
+        var self = this; 
+        return self._transporter.promise;
+    }
+
+    _loadStatus(){
+        var self = this;
+        return self._status.promise;
+    }
+
+    _loadSendCheck(){
```

Comment:
As mentioned in my earlier comment, add the proper comments in the correct format for this and every other method